HLW8032 documentation (#4124)
OttoWinter/esphomedocs: content/components/sensor/hlw8032.md

---
description: "Instructions for setting up HLW8032 power sensors"
title: "HLW8032 Power Sensor"
params:
  seo:
    description: Instructions for setting up HLW8032 power sensors
    image: hlw8032.jpg
---

> The `hlw8032`   sensor platform allows you to use your HLW8032 voltage/current and power sensors

Reviewer: ```suggestion
The `hlw8032` sensor platform allows you to use your HLW8032 voltage/current and power sensors
```

> ([datasheet](https://v4.cecdn.yun300.cn/100001_2107285051/DS_HLW8032_EN_Rev1.5.pdf)) sensors with

Reviewer: ```suggestion
The `hlw8032` sensor platform allows you to use your HLW8032 voltage/current and power sensors
([datasheet](https://v4.cecdn.yun300.cn/100001_2107285051/DS_HLW8032_EN_Rev1.5.pdf)) with
```

> ESPHome. This sensor is commonly found in M5Stack modules.

Reviewer: ```suggestion
The `hlw8032` sensor platform allows you to use your HLW8032 voltage/current and power sensors
([datasheet](https://v4.cecdn.yun300.cn/100001_2107285051/DS_HLW8032_EN_Rev1.5.pdf)) with
ESPHome. This sensor is commonly found in M5Stack modules.
```

> [!WARNING]
> SAFETY HAZARD: Some devices have the digital GND connected directly to mains
> voltage so **the GPIOs become LIVE during normal operation**. Our advice is to mark these boards to prevent
> any use of the dangerous digital pins. UART TX pin is usually connected to the MCU via optocoupler.

The HLW8032 IC measures a single phase's voltage (using a voltage divider) and current (using a shunt)
and additionally provides active and apparent power and power factor measurements.

As the communication with the HLW8032 done using UART, you need

> to have an [UART bus](#uart) in your configuration with the `rx_pin` connected to the HLW8032.

Reviewer: ```suggestion
As the communication with the HLW8032 is done using UART, you need
to have a [UART bus](#uart) in your configuration with the `rx_pin` connected to the HLW8032.
```

> to have an [UART bus](#uart) in your configuration with the `rx_pin` connected to the HLW8032.

Reviewer: ```suggestion
As the communication with the HLW8032 is done using UART, you need
to have a [UART bus](#uart) in your configuration with the `rx_pin` connected to the HLW8032.
```

```yaml
# Example configuration entry
uart:
  rx_pin: GPIOXX
  baud_rate: 4800
  parity: EVEN

sensor:
  - platform: hlw8032
    voltage:
      name: HLW8032 Voltage
      id: hlw8032_voltage
    current:
      name: HLW8032 Current
      id: hlw8032_current
    power:
      name: HLW8032 Power
      id: hlw8032_power
    apparent_power:
      name: HLW8032 Apparent Power
      id: hlw8032_apparent_power
    power_factor:
      name: HLW8032 Power Factor
      id: hlw8032_power_factor
```

## Configuration variables

- **current** (*Optional*): Use the current value of the sensor in amperes.
  All options from [Sensor](#config-sensor).
- **power** (*Optional*): Use the (active) power value of the sensor in watts.
  All options from [Sensor](#config-sensor).
- **voltage** (*Optional*): Use the voltage value of the sensor in V (RMS).
  All options from [Sensor](#config-sensor).
- **apparent_power** (*Optional*): Use the apparent power value of the sensor in volt amps.
  All options from [Sensor](#config-sensor).
- **power_factor** (*Optional*): Use the power factor value of the sensor.

>   All options from [Sensor](#config-sensor).

Reviewer: ```suggestion
  All options from [Sensor](#config-sensor).
- **uart_id** (*Optional*, [ID](#config-id)): Manually specify the ID of the [UART Component](#uart) if you want
  to use multiple UART buses.
```

Advanced Options:

- **current_resistor** (*Optional*, float): The value of the shunt resistor for current measurement.
  Default value is `0.001 ohm`.
- **voltage_divider** (*Optional*, float): The value of the voltage divider
  on the board as `R_upstream / (R_downstream * 1000)`. Default value is `1.72`.

## See Also

> - [sensor-filters](#sensor-filters)

Reviewer: ```suggestion
- [Sensor Filters](#sensor-filters)
```

> - [sensor-filters](#sensor-filters)

Reviewer: ```suggestion
- [Sensor Filters](#sensor-filters)
```

---
description: "Instructions for setting up HLW8032 power sensors"
title: "HLW8032 Power Sensor"
params:
  seo:
    description: Instructions for setting up HLW8032 power sensors
    image: hlw8032.jpg
---

The `hlw8032` sensor platform allows you to use your HLW8032 voltage/current and power sensors
([datasheet](https://v4.cecdn.yun300.cn/100001_2107285051/DS_HLW8032_EN_Rev1.5.pdf)) with
ESPHome. This sensor is commonly found in M5Stack modules.

> [!WARNING]
> SAFETY HAZARD: Some devices have the digital GND connected directly to mains
> voltage so **the GPIOs become LIVE during normal operation**. Our advice is to mark these boards to prevent
> any use of the dangerous digital pins. UART TX pin is usually connected to the MCU via optocoupler.

The HLW8032 IC measures a single phase's voltage (using a voltage divider) and current (using a shunt)
and additionally provides active and apparent power and power factor measurements.

As the communication with the HLW8032 is done using UART, you need

> to have a [UART bus](#uart) in your configuration with the `rx_pin` connected to the HLW8032.

Reviewer: Hi, sorry I forgot about this PR and would like to merge in the next few days. Can you merge the latest or allow me to make changes? There were some updates and this will fail lint.

---
description: "Instructions for setting up HLW8032 power sensors"
title: "HLW8032 Power Sensor"
params:
  seo:
    description: Instructions for setting up HLW8032 power sensors
    image: hlw8032.jpg
---

The `hlw8032`   sensor platform allows you to use your HLW8032 voltage/current and power sensors
([datasheet](https://v4.cecdn.yun300.cn/100001_2107285051/DS_HLW8032_EN_Rev1.5.pdf)) sensors with
ESPHome. This sensor is commonly found in M5Stack modules.

> [!WARNING]
> SAFETY HAZARD: Some devices have the digital GND connected directly to mains
> voltage so **the GPIOs become LIVE during normal operation**. Our advice is to mark these boards to prevent
> any use of the dangerous digital pins. UART TX pin is usually connected to the MCU via optocoupler.

The HLW8032 IC measures a single phase's voltage (using a voltage divider) and current (using a shunt)
and additionally provides active and apparent power and power factor measurements.

As the communication with the HLW8032 done using UART, you need
to have an [UART bus](#uart) in your configuration with the `rx_pin`   connected to the HLW8032.

> Additionally, you need to set the baud rate to `4800`   and the `EVEN`   parity.  The device sends multiple updates per second, so you

Reviewer: Lines should be less than 120 cols long

will probably want some sort of averaging or throttle filter on the sensors.

```yaml
# Example configuration entry
uart:

>   rx_pin: RX

Reviewer: ```suggestion
  rx_pin: GPIOXX
```

  baud_rate: 4800
  parity: EVEN

sensor:
  - platform: hlw8032
    voltage:
      name: HLW8032 Voltage
      id: hlw8032_voltage
    current:
      name: HLW8032 Current
      id: hlw8032_current
    power:
      name: HLW8032 Power
      id: hlw8032_power
    apparent_power:
      name: HLW8032 Apparent Power
      id: hlw8032_apparent_power
    power_factor:
      name: HLW8032 Power Factor
      id: hlw8032_power_factor

> 

Reviewer: Blank line

```

## Configuration variables

- **current** (*Optional*): Use the current value of the sensor in amperes.
  All options from [Sensor](#config-sensor).
- **power** (*Optional*): Use the (active) power value of the sensor in watts.
  All options from [Sensor](#config-sensor).
- **voltage** (*Optional*): Use the voltage value of the sensor in V (RMS).
  All options from [Sensor](#config-sensor).
- **apparent_power** (*Optional*): Use the apparent power value of the sensor in volt amps.
  All options from [Sensor](#config-sensor).
- **power_factor** (*Optional*): Use the power factor value of the sensor.
  All options from [Sensor](#config-sensor).

Advanced Options:

- **current_resistor** (*Optional*, float): The value of the shunt resistor for current measurement.
  Defaults value `0.001 ohm`  .
- **voltage_divider** (*Optional*, float): The value of the voltage divider on the board as `R_upstream / (R_downstream * 1000)`  .

>   Defaults value `1.72`  .

Reviewer: Extra spaces before .

Reviewer: 'Defaults value' should be changed to 'Default value'
